src/Client.php

```diff
@@ -219,7 +225,10 @@ public function getAdapter()
             $this->setAdapter($this->config['adapter']);
         }
 
-        return $this->adapter;
+        /** @var Client\Adapter\AdapterInterface $adapter */
+        $adapter = $this->adapter;
+
+        return $adapter;
```

Comment:
I think we can't say that `$adapter` can only be an `AdapterInterface` because is not set in constructor. `$adapter` will be `null` until some method will call `getAdapter()`, then a developer should know that `$this->adapter` can be `null`.

Maybe, can we initialize it in constructor?